python/pyspark/ml/wrapper.py

```diff
@@ -48,12 +48,13 @@ def __init__(self, java_obj: Optional["JavaObject"] = None):
         self._java_obj = java_obj
 
     def __del__(self) -> None:
-        from pyspark.core.context import SparkContext
-
-        if SparkContext._active_spark_context and self._java_obj is not None:
-            SparkContext._active_spark_context._gateway.detach(  # type: ignore[union-attr]
-                self._java_obj
-            )
+        # TODO remove the object on the server side
+        if self._java_obj is not None and not is_remote():
```

Comment:
What's our logic to remove cached objects on server?

Reply:
If the model object in python side has been GCed, then it will launch a GRPC call to delete the cached model in the server.

Reply:
Should we have such logic in this section?

Reply:
Not yet. will support it.